client/my-sites/plugins/index.js

```diff
@@ -82,8 +82,12 @@ module.exports = function() {
 			pluginsController.eligibility
 		);
 
-		page.exit( '/plugins*',
-			pluginsController.resetHistory
-		);
+		page.exit( '/plugins*', ( context, next ) => {
+			if ( 0 !== page.current.indexOf( '/plugins' ) ) {
```

Comment:
Should we change this and the path pattern to include the trailing `/` in case we ever decide to add unrelated  `/plugins-foo` routes in the future?

I kinda hoped there'd be an easy way to reuse the regular expression generated by page.js here to avoid a string comparison on `'/plugins'` but it doesn't seem there's an easy way to get at that. At least not without some horrendous abuse of internal APIs such as:

```js
const exitRoute = new page.Route( '/plugins*' );
page.exits.push( exitRoute.middleware( ( context, next ) => {
	if ( ! exitRoute.regexp.test( page.current ) ) {
		pluginsController.resetHistory();
	}

	next();
} ) );
```

(The [page.js source](https://github.com/visionmedia/page.js/blob/master/page.js) is surprisingly approachable)